apis/api/commontypes/astarte_common_types.go

```diff
@@ -344,6 +344,14 @@ type AstarteVerneMQSpec struct {
 	PersistentClientExpiration string `json:"persistentClientExpiration,omitempty"`
 	// +optional
 	MirrorQueue string `json:"mirrorQueue,omitempty"`
+	// This option allows, when true, to handle SSL termination at VerneMQ level.
+	// Default: false
+	// +optional
+	SSLListener *bool `json:"sslListener,omitempty"`
+	// Reference to the secret containing the TLS certificate for VerneMQ.
+	// The field will be used only if SSLListener is set to true.
+	// +optional
+	SSLListenerSecretRef *v1.SecretReference `json:"sslListenerSecretRef,omitempty"`
```

Comment:
I think this is unnecessarily verbose, use `SSLListenerCertificate` or something like this, we should rather specify the purpose than the type (especially given it is strongly typed)